src/tools/recipe2plan.ts

```diff
@@ -0,0 +1,162 @@
+   * @throws Error if recipe fails to resolve on first or second pass.
+   * @yields Resolved recipes with storage keys
+   */
+  async* resolve(): AsyncIterator<Recipe> {
+    for (const r of this.runtime.context.allRecipes) {
+      const arc = this.runtime.newArc(this.getArcId(r), ramDiskStorageKeyPrefixForTest());
+      const opts = {errors: new Map<Recipe | RecipeComponent, string>()};
+      const resolved = await this.resolveOrNormalize(r, arc, opts);
+      if (!resolved) {
+        throw Error(`Recipe ${r.name} failed to resolve:\n${[...opts.errors.values()].join('\n')}`);
+      }
+      this.createStoresForCreateHandles(resolved, arc);
+      resolved.normalize();
+      if (!resolved.isResolved()) {
+        throw Error(`Recipe ${resolved.name} did not properly resolve!\n${resolved.toString({showUnresolved: true})}`);
+      }
+      yield resolved;
+    }
+  }
+
+  /**
+   * Resolves unresolved recipe or normalizes resolved recipe.
+   *
+   * @param recipe long-running or ephemeral recipe
+   * @param arc Arc is associated with input recipe
+   * @param opts contains `errors` map for reporting.
+   */
+  async resolveOrNormalize(recipe: Recipe, arc: Arc, opts?: IsValidOptions): Promise<Recipe | null> {
+    const normed = recipe.clone();
+    normed.normalize();
+    if (normed.isResolved()) return normed;
+
+    return await (new RecipeResolver(arc).resolve(recipe, opts));
+  }
+
+  /** @returns the arcId from annotations on the recipe when present. */
+  getArcId(recipe: Recipe): string | null {
+    const triggers: [string, string][][] = recipe.triggers;
+    for (const trigger of triggers) {
+      for (const pair of trigger) {
+        if (pair[0] === 'arcId') {
+          return pair[1];
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Create stores with keys for all create handles.
+   *
+   * @param recipe should be long running.
+   * @param arc Arc is associated with current recipe.
+   */
+  createStoresForCreateHandles(recipe: Recipe, arc: Arc) {
+    const resolver = new CapabilitiesResolver({arcId: arc.id});
+    for (const ch of recipe.handles.filter(h => h.fate === 'create')) {
+      const storageKey = ramDiskStorageKeyPrefixForTest()(arc.id); // TODO: actually create the storage keys.
+      const store = new Store({storageKey, exists: Exists.MayExist, type: ch.type, id: ch.id});
+      arc.context.registerStore(store, ch.tags);
+    }
+  }
+
+  /**
+   * WIP method to match `map` and `copy` fated handles with storage keys from `create` handles.
+   *
+   * @throws when a mapped handle is associated with too many stores (ambiguous mapping).
+   * @throws when a mapped handle isn't associated with any store (no matches found).
+   * @throws when handle is mapped to a handle from an ephemeral recipe.
+   * @param recipe long-running or ephemeral recipe
+   */
+  matchKeysToHandles(recipe: Recipe) {
+    recipe.handles
+      .filter(h => h.fate === 'map' || h.fate === 'copy')
+      .forEach(h => {
```

Comment:
This lambda is long enough to warrant spelling out the variable name properly: `handle => {`